semantics/src/traversal/assignments.rs

```diff
@@ -53,11 +54,15 @@ fn assign_subscript(
         slices,
     } = &target.node
     {
-        let _target_attributes = expressions::expr(Rc::clone(&scope), Rc::clone(&context), target)?;
-        let _value_attributes = expressions::expr(Rc::clone(&scope), Rc::clone(&context), value)?;
-        let _index_attributes = expressions::slices_index(scope, context, slices)?;
+        let target_attributes = expressions::expr(Rc::clone(&scope), Rc::clone(&context), target)?;
+        let value_attributes = expressions::expr(Rc::clone(&scope), Rc::clone(&context), value)?;
+        let index_attributes = expressions::slices_index(scope, context, slices)?;
 
-        // TODO: perform type checking
+        let indexed_type = operations::index(target_attributes.typ, index_attributes.typ)?;
+
+        if indexed_type != value_attributes.typ {
+            return Err(SemanticError::TypeError);
```

Comment:
Not for this PR but should the different branches of `SemanticError` allow to carry a string to be able to give the user better feedback about the specific type error?

Reply:
Yeah, I'd like to add this at some point. My thinking was to build the semantic pass out a bit more and then find a good approach to handling errors. It would also be nice to include span data and have it so that multiple errors can be returned.

Reply:
>it would also be nice to include span data and have it so that multiple errors can be returned.

Yep, that sounds good.